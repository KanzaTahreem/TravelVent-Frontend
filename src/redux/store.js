import { configureStore } from '@reduxjs/toolkit';
import reserveItems from './reserveSlice';
// import cartItems from './cartSlice';
import tripSlice from './tripSlice';
import sessionSlice from './reducers/sessionSlice';

const store = configureStore({
  reducer: {
    reserveItems,
    trips: tripSlice,
    session: sessionSlice,
  },
});

export default store;
