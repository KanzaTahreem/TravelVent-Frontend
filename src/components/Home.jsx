import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchTrips } from '../redux/tripSlice';

import DotSeperator from './DotSeperator';
import '../assets/stylesheets/home.css';

const Home = () => {
  const trips = useSelector((state) => state.trips.trips);
  const dispatch = useDispatch();
  const [startIndex, setStartIndex] = useState(0);
  const [endIndex, setEndIndex] = useState(2);

  useEffect(() => {
    dispatch(fetchTrips());
  }, [dispatch]);

  const handleClickLeft = () => {
    if (startIndex > 0) {
      setStartIndex(startIndex - 3);
      setEndIndex(endIndex - 3);
    }
  };

  const handleClickRight = () => {
    if (endIndex < trips.length - 1) {
      setStartIndex(startIndex + 3);
      setEndIndex(endIndex + 3);
    }
  };

  const headerDotStyle = {
    marginTop: '25px',
  };

  const itemDotStyle = {
    marginBottom: '10px',
    fontSize: '12px',
  };

  return (
    <div className="column home-page">
      <div className="home-page-header column">
        <h1 className="home-title">Explore the world</h1>
        <p className="home-subtitle">Find your next adventure</p>
        <DotSeperator style={headerDotStyle}>
          ..........................
        </DotSeperator>
      </div>

      <div className="trip-cards-container row">
        {trips.slice(startIndex, endIndex + 1).map((trip) => (
          <Link to={`/trips/${trip.id}`} key={trip.id} className="trip-card">
            <img src={trip.image_url} alt={trip.destination_city} className="home-img" />
            <div className="trip-info">
              <h2>{trip?.destination_city.toUpperCase()}</h2>
              <DotSeperator style={itemDotStyle}>
                ............
              </DotSeperator>
              <p className="home-trip-description">{trip.description}</p>
            </div>
          </Link>
        ))}

      </div>
      <button type="button" className="arrow-button left-arrow" onClick={handleClickLeft}>
        <i className="fa fa-angle-left" />
      </button>

      <button type="button" className="arrow-button right-arrow" onClick={handleClickRight}>
        <i className="fa fa-angle-right" />
      </button>
    </div>
  );
};

export default Home;
