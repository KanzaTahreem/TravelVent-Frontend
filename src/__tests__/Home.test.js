import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Home from '../components/Home';

describe('Testing Home', () => {
  let home;
  beforeAll(async () => {
    home = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Home />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  it('should renders correctly', () => {
    expect(home).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(home).toMatchSnapshot();
  });

  it('renders the component', () => {
    waitFor(() => expect(screen.getByText('Explore the world')).toBeInTheDocument());
    waitFor(() => expect(screen.getByText('Find your next adventure')).toBeInTheDocument());
    waitFor(() => expect(screen.getAllByTestId('trip-card')).toHaveLength(3));
  });

  it('Should have a link to the details page', () => {
    const link = screen.findByTestId('trip-card');
    waitFor(() => expect(link).toBeInTheDocument());
    waitFor(() => expect(link).toHaveAttribute('href'));
  });
});
