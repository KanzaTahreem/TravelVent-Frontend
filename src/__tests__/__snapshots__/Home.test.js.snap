// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home should match the snapshot 1`] = `
<div
  className="column home-page"
>
  <div
    className="home-page-header column"
  >
    <h1
      className="home-title"
    >
      Explore the world
    </h1>
    <p
      className="home-subtitle"
    >
      Find your next adventure
    </p>
    <p
      className="dot-seperator"
      style={
        Object {
          "marginTop": "25px",
        }
      }
    >
      ..........................
    </p>
  </div>
  <div
    className="trip-cards-container row"
  />
  <button
    className="arrow-button left-arrow"
    data-testid="left-arrow"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10 12.796V3.204L4.519 8 10 12.796zm-.659.753-5.48-4.796a1 1 0 0 1 0-1.506l5.48-4.796A1 1 0 0 1 11 3.204v9.592a1 1 0 0 1-1.659.753z"
      />
    </svg>
  </button>
  <button
    className="arrow-button right-arrow"
    data-testid="right-arrow"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "grey",
      }
    }
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 12.796V3.204L11.481 8 6 12.796zm.659.753 5.48-4.796a1 1 0 0 0 0-1.506L6.66 2.451C6.011 1.885 5 2.345 5 3.204v9.592a1 1 0 0 0 1.659.753z"
      />
    </svg>
  </button>
</div>
`;
